packages/CalendarCarousel/index.tsx

```diff
@@ -1,19 +1,274 @@
+  beforeArrowText: TextStyle;
+  titleFlex: ViewStyle;
+  titleText: TextStyle;
+  yearText: TextStyle;
+  nextArrowText: TextStyle;
+  weekdayFlex: ViewStyle;
+  weekdayText: TextStyle;
+  calendarContainerSize: ViewStyle;
+}
 
-const styles = StyleSheet.create({
+const styles = StyleSheet.create<Style>({
   container: {
     flex: 1,
+    paddingTop: 40,
+    width: 330,
+    height: 388,
+  },
+  headerFlex: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    paddingBottom: 30,
+  },
+  beforeArrowText: {
+    color: 'royalblue',
+    fontSize: 30,
+  },
+  titleFlex: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  titleText: {
+    fontFamily: 'Avenir',
+    fontSize: 20,
+    textAlign: 'center',
+    justifyContent: 'center',
+    width: 300,
+  },
+  yearText: {
+    fontSize: 12,
+    textAlign: 'center',
+    justifyContent: 'center',
+  },
+  nextArrowText: {
+    color: 'royalblue',
+    fontSize: 30,
+  },
+  weekdayFlex: {
+    flexDirection: 'row',
+  },
+  weekdayText: {
+    color: '#4F4F4F',
+    fontFamily: 'Avenir',
+    fontSize: 20,
+    textAlign: 'center',
+  },
+  calendarContainerSize: {
+    width: 330,
+    height: 388,
   },
 });
 
-function Item(): React.ReactElement {
+const data = {
+  dates: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+  week: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
+  month: [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ],
+  Day: new Date().getDate(),
+};
+
+interface Props<T> {
+  month?: number;
+  onPrevMonth?: () => void;
+  onNextMonth?: () => void;
+  swipeLeft?: () => void;
+  swipeRight?: () => void;
+}
+
+function CalendarCarousel<T>(props: Props<T>): React.ReactElement {
+  const { month, onPrevMonth, onNextMonth, swipeLeft, swipeRight } = props;
+  const firstDay = new Date(new Date().getFullYear(), month, 1).getDay();
+  const year = new Date().getFullYear();
+  const [isSwapping, setIsSwapping] = React.useState(false);
+
+  if (((year % 4 && year % 100) || year % 400 === 0) !== 0) {
+    const dates = (data.dates[1] = 29);
+  } else {
+    const dates = data.dates;
+  }
+
+  const weekdays = [];
+  for (let idx = 0; idx <= 6; idx++) {
+    const d = data.week[idx];
+    weekdays.push(
+      <View style={{ width: 47.14 }} key={idx}>
+        <Text style={styles.weekdayText}>{d}</Text>
+      </View>,
+    );
+  }
+
+  const currentMonth = data.month[month];
+  const numDays = data.dates[month];
+
+  let prevDate = data.dates[month - 1];
+  if (month === 0) {
+    prevDate = data.dates[11];
+  }
+
+  const prevMonthDays = [];
+  for (let idx = 0; idx < 10; idx++) {
+    prevMonthDays.push(prevDate);
+    prevDate--;
+  }
+
+  const blanks = [];
+  for (let i = 0; i < firstDay; i++) {
+    blanks.push(i);
+  }
+  const numBlanks = blanks.length;
+  const blankDays = prevMonthDays.slice(0, numBlanks).reverse();
```

Comment:
Reverse again? I think this can be written in a better form. Such as making one `for` loop for adding `prevMonthDays`